shell/platform/linux/fl_view.cc

```diff
@@ -255,12 +271,135 @@ static void fl_view_size_allocate(GtkWidget* widget,
   FlutterEngineSendWindowMetricsEvent(self->flutter_engine, &event);
 }
 
+static gboolean fl_view_button_press_event(GtkWidget* widget,
+                                           GdkEventButton* event) {
+  FlView* self = FL_VIEW(widget);
+
+  if (self->flutter_engine == nullptr)
+    return FALSE;
+
+  // Flutter doesn't handle double and triple click events
+  if (event->type == GDK_DOUBLE_BUTTON_PRESS ||
+      event->type == GDK_TRIPLE_BUTTON_PRESS)
+    return FALSE;
+
+  int64_t button = gdk_button_to_flutter_button(event->button);
+  if (button < 0)
+    return FALSE;
+
+  // Drop the event if Flutter already thinks the button is down
+  if ((self->button_state & button) != 0)
+    return FALSE;
+  self->button_state ^= button;
+
+  FlutterPointerEvent fl_event = {};
+  fl_event.struct_size = sizeof(fl_event);
+  fl_event.phase = kDown;
```

Comment:
Per my other comment, this will need to be something like `= any_button_was_down ? kMove : kDown` (where that's set by `self->button_state != 0` before the state update above).